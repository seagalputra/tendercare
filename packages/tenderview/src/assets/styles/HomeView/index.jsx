import { makeStyles } from '@material-ui/core/styles'

const useStyles = makeStyles(theme => ({
  paper: {
    marginTop: theme.spacing(8),
    [theme.breakpoints.down('xs')]: {
      marginTop: theme.spacing(0)
    },
    display: 'flex',
    flexDirection: 'column',
    alignItems: 'center',
    padding: theme.spacing(6)
  },
  image: {
    width: '78%',
    margin: 'auto'
  }
}))

export default useStyles
