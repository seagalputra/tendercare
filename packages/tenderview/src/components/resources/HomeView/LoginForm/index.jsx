import React, { useState } from 'react'
import {
  Button,
  TextField,
  FormControlLabel,
  Checkbox,
  Link,
  Typography,
  CircularProgress,
  Grid
} from '@material-ui/core'
import useStyles from 'assets/styles/LoginForm'
import { useFormik } from 'formik'
import * as Yup from 'yup'

import authenticateUser from 'api/user'
import { storeToken } from 'utils/sessions'

const validationForm = Yup.object({
  email: Yup.string()
    .email('Masukkan alamat email yang valid')
    .required(),
  password: Yup.string().required()
})

const LoginForm = ({ handleRedirect }) => {
  const [isLoading, setIsLoading] = useState(false)
  const classes = useStyles()

  const handleLogin = async request => {
    const { email, password } = request
    setIsLoading(true)
    const {
      data: { token }
    } = await authenticateUser(email, password)

    storeToken(token)
    handleRedirect('/dashboard')
  }

  const formik = useFormik({
    initialValues: {
      email: '',
      password: ''
    },
    validationSchema: validationForm,
    onSubmit: data => handleLogin(data)
  })

  return (
    <>
      <Typography className={classes.title} variant="h5" gutterBottom>
        Selamat Datang di SPO 1047
      </Typography>
      <Typography variant="subtitle1">
        Silahkan masuk untuk melanjutkan
      </Typography>

      <form className={classes.form} onSubmit={formik.handleSubmit}>
        <TextField
          error={formik.touched.email && Boolean(formik.errors.email)}
          variant="outlined"
          margin="normal"
          required
          fullWidth
          id="email"
          label="Alamat Email"
          name="email"
          autoComplete="email"
          type="email"
          autoFocus
          onChange={formik.handleChange}
          onBlur={formik.handleBlur}
          value={formik.values.email}
        />
        <TextField
          error={formik.touched.password && Boolean(formik.errors.password)}
          variant="outlined"
          margin="normal"
          required
          fullWidth
          id="password"
          label="Kata Sandi"
          name="password"
          type="password"
          onChange={formik.handleChange}
          onBlur={formik.handleBlur}
          value={formik.values.password}
        />
        <FormControlLabel
          control={<Checkbox value="remember" color="primary" />}
          label="Ingat Saya"
        />
        {isLoading ? (
          <Button
            fullWidth
            variant="contained"
            disabled
            className={classes.submit}
          >
            <CircularProgress size={24} color="primary" />
          </Button>
        ) : (
          <Button
            type="submit"
            fullWidth
            variant="contained"
            color="primary"
            className={classes.submit}
          >
            Masuk
          </Button>
        )}
        <Grid container>
          <Grid item xs>
            <Link href="!#" variant="body2">
              Lupa Password?
            </Link>
          </Grid>
          <Grid item>
            <Link href="/register" variant="body2">
              Tidak punya akun? Daftar
            </Link>
          </Grid>
        </Grid>
      </form>
    </>
  )
}

export default LoginForm
