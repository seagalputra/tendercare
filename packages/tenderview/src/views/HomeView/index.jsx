import React from 'react'
import { Grid, Hidden } from '@material-ui/core'
import { useHistory } from 'react-router-dom'

import LoginForm from 'components/resources/HomeView/LoginForm'

import useStyles from 'assets/styles/HomeView'

import backgroundImage from 'assets/img/delivery.svg'

const HomeView = () => {
  const classes = useStyles()
  const history = useHistory()

  const handleRedirect = path => {
    history.push(path)
  }

  return (
    <Grid container component="main">
      <Hidden smDown>
        <Grid
          container
          item
          xs={false}
          sm={false}
          md={7}
          justify="center"
          alignItems="center"
        >
          <img
            src={backgroundImage}
            alt="Background"
            className={classes.image}
          />
        </Grid>
      </Hidden>
      <Grid item xs={12} sm={12} md={5} className={classes.paper}>
        <LoginForm handleRedirect={handleRedirect} />
      </Grid>
    </Grid>
  )
}

export default HomeView
