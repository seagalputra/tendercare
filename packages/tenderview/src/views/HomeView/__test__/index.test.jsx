import React from 'react'
import { shallow } from 'enzyme'
import HomeView from '..'

jest.mock('react-router-dom', () => ({
  ...jest.requireActual('react-router-dom'),
  useHistory: () => ({
    push: jest.fn()
  })
}))

describe('HomeView', () => {
  it('renders without crashing', () => {
    const component = shallow(<HomeView />)
    expect(component).toMatchSnapshot()
  })
})
