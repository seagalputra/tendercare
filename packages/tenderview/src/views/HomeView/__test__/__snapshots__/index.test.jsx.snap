// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HomeView renders without crashing 1`] = `
<WithStyles(ForwardRef(Grid))
  component="main"
  container={true}
>
  <Hidden
    smDown={true}
  >
    <WithStyles(ForwardRef(Grid))
      alignItems="center"
      container={true}
      item={true}
      justify="center"
      md={7}
      sm={false}
      xs={false}
    >
      <img
        alt="Background"
        className="makeStyles-image-2"
        src="delivery.svg"
      />
    </WithStyles(ForwardRef(Grid))>
  </Hidden>
  <WithStyles(ForwardRef(Grid))
    className="makeStyles-paper-1"
    item={true}
    md={5}
    sm={12}
    xs={12}
  >
    <LoginForm
      handleRedirect={[Function]}
    />
  </WithStyles(ForwardRef(Grid))>
</WithStyles(ForwardRef(Grid))>
`;
